src/machine/machine_nrf.go

```diff
@@ -333,3 +334,36 @@ func (i2c *I2C) readByte() (byte, error) {
 	i2c.Bus.EVENTS_RXDREADY.Set(0)
 	return byte(i2c.Bus.RXD.Get()), nil
 }
+
+var rngInitDone = false
+
+const RNG_MAX_READ_RETRIES = 1000
+
+// GetRNG returns 32 bits of cryptographically secure random data
+func GetRNG() (uint32, error) {
+	if !rngInitDone {
+		nrf.RNG.CONFIG.Set(nrf.RNG_CONFIG_DERCEN)
+		nrf.RNG.SHORTS.ClearBits(nrf.RNG_SHORTS_VALRDY_STOP)
```

Comment:
This doesn't work together with the SoftDevice. Instead of using the hardware directly, we should call the SoftDevice APIs. Please take a look here:
https://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.s132.api.v7.0.1%2Fgroup___n_r_f___s_o_c___f_u_n_c_t_i_o_n_s.html&cp=4_7_3_3_2_7_2_41&anchor=ga825412a3746ef587c60d221240691dcd

Reply:
For an example how this can be done, please take a look here: https://github.com/tinygo-org/tinygo/blob/release/src/runtime/runtime_nrf_softdevice.go

Reply:
This is where I got the sequence I used https://github.com/SimpleHacks/hw_rng/blob/master/src/hw_rng.h#L75

But you are no doubt correct, although a bit more complex to correctly implement.

Reply:
Where can I lookup the SVCall numbers?

Reply:
I found them: https://github.com/tinygo-org/bluetooth/blob/release/s110_nrf51_8.0.0/s110_nrf51_8.0.0_API/include/nrf_soc.h#L119